Code/Editor/AzAssetBrowser/AzAssetBrowserRequestHandler.cpp

```diff
@@ -95,109 +100,151 @@ namespace AzAssetBrowserRequestHandlerPrivate
-                {
-                    newEntity->RemoveComponent(newComponent);
-                    newComponent = aznew AzToolsFramework::Components::GenericComponentWrapper(newComponent);
-                    newEntity->AddComponent(newComponent);
-                }
+        // If not sourced from an asset, generate a generic name.
+        if (entityName.empty())
+        {
+            entityName = AZStd::string::format("Entity%d", GetIEditor()->GetObjectManager()->GetObjectCount());
+        }
 
-                newEntity->Activate();
+        AZ::EntityId targetEntityId;
+        EditorRequests::Bus::BroadcastResult(targetEntityId, &EditorRequests::CreateNewEntityAtPosition, worldTransform.GetTranslation(), AZ::EntityId());
 
-                // set asset after components have been activated in AddEditorEntity method
-                if (newComponent)
-                {
-                    Components::EditorComponentBase* asEditorComponent =
-                        azrtti_cast<Components::EditorComponentBase*>(newComponent);
+        AZ::Entity* newEntity = nullptr;
+        AZ::ComponentApplicationBus::BroadcastResult(newEntity, &AZ::ComponentApplicationRequests::FindEntity, targetEntityId);
 
-                    if (asEditorComponent)
-                    {
-                        asEditorComponent->SetPrimaryAsset(assetId);
-                    }
-                }
+        if (newEntity == nullptr)
+        {
+            QMessageBox::warning(mainWindow, QObject::tr("Asset Drop Failed"), QStringLiteral("Could not create entity from selected asset(s)."));
+            return;
+        }
+
+        // Deactivate the entity so the properties on the components can be set.
+        newEntity->Deactivate();
+
+        newEntity->SetName(entityName);
+
+
+        // Add all dropped products as components to this entity.
+        AZStd::vector<AZ::EntityId> entityIds = { targetEntityId };
+        EntityCompositionRequests::AddComponentsOutcome addComponentsOutcome = AZ::Failure(AZStd::string());
+        EntityCompositionRequestBus::BroadcastResult(
+            addComponentsOutcome, &EntityCompositionRequests::AddComponentsToEntities, entityIds, componentsToAdd);
 
-                bool isPrefabSystemEnabled = false;
-                AzFramework::ApplicationRequests::Bus::BroadcastResult(
-                    isPrefabSystemEnabled, &AzFramework::ApplicationRequests::IsPrefabSystemEnabled);
+        if (!addComponentsOutcome.IsSuccess())
+        {
+            QMessageBox::warning(mainWindow, QObject::tr("Asset Drop Failed"), QStringLiteral("Could not create components for entity from selected asset(s)."));
+
+            EditorEntityContextRequestBus::Broadcast(&EditorEntityContextRequests::DestroyEditorEntity, targetEntityId);
+            return;
+        }
 
-                if (!isPrefabSystemEnabled)
+        
+        for (const AssetIdAndComponentTypeId& assetAndComponent : assetIdAndComponentTypeId)
```

Comment:
Not quite, it assigns it to the first component with that ID. So if your products all have different component IDs it will work as you would expect. If you have multiple products with the same component ID it does what you expect and doesn't assign it correctly. However, this bug exists in dropping to the outliner as well, and I just wanted to match behavior here, not fix the bug.